### PR TITLE
 #169257395  Builds Users' verification emails using a 3rd party api

### PR DESCRIPTION
**Pull request by b0nbon1**
#### What does this PR do?
This PR verifies user emails by sending them an email which they have to confirm so as to finish their registration during signing up.
#### Description of Task to be completed?
- write tests
- install Sendgrid which is 3rd party API for sending emails
- Create an email template
- add endpoint `POST /api/v1/auth/verification`
- refractor test to make sure they are passing
#### How should this be manually tested?
- git clone the repo
- create a Sendgrid account and get the API key
- add the Sendgrid API key to .env
- run `npm install` to install the dependencies
#### option 1
- run `npm run migrate` to make migrations
- run `npm run watch` to run the server in debug mode
- go to postman 
- on endpoint ` Post http://localhost:3000/api/v1/auth/signup` create a new user with an email that exists eg.
```json
{
	"firstName": "new",
	"lastName": "user",
	"email": "validuser@email.com",
	"password": "1234abcd"
}
```
- check email used, for the verification link
- click the link to verify the user
#### option 2
- run `npm test` to run the tests
#### Any background context you want to provide?
- this PR depends on register new user so it must be merged so for it to work, meanwhile its working under dummy create user endpoint
- [Implementation plan](https://docs.google.com/document/d/1x1LayxqF7LSHr0GlsFYWGC6r6Pw-WqYGxKFI7pIujzg/edit?usp=sharing)
#### What are the relevant pivotal tracker stories?
- [#169257395](https://www.pivotaltracker.com/n/projects/2407417/stories/169257395)
#### Screenshots:
- email template:
Screen Shot 2019-11-15 at 07 59 14

- email verification endpoint:
image

#### Questions:
- Can I use pug to template the emails sent and how?
- How will the template look like

### CODE REVIEW

File: src/services/user.service.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+   * @param {string} email
+   * @returns {object} user
+   */
+  static async findByEmail(email) {
+    const user = await db.user.findOne({
+      where: { email }
+    });
+    if (!user) return null;
+    return user;
+  }
```
Comment (nshaphan) [inline src/services/user.service.js]: It would be a good idea if this was in the users' controller.

**Review by gildniy — DISMISSED** [review]
Looks great for me!

**Review by nshaphan — DISMISSED** [review]
Great job @b0nbon1

File: src/services/user.service.js (hunk 1 above)
Comment (b0nbon1) [reply src/services/user.service.js]: Okay, done. Thanks for that

**Review by RaymondMwaura — COMMENTED** [review]
This didn't work for me and yet I have added my API key?!

**Comment by b0nbon1** [comment]
what is the issue @RaymondMwaura

**Comment by b0nbon1** [comment]
> This didn't work for me and yet I have added my API key?!

I think we have fixed this @RaymondMwaura

**Review by RaymondMwaura — DISMISSED** [review]
Great job!

**Review by timolinn — COMMENTED** [review]
Test failing with `ERROR: Unable to find migration: 20191104140247-create-user.js`

**Comment by b0nbon1** [comment]
> Test failing with `ERROR: Unable to find migration: 20191104140247-create-user.js`

I think that migration was recreated and there is a new migration. because I got the changes from @nshaphan  which had new migration `users` and the old migration had been deleted. 
@timolinn  The best solution is to drop all the tables manually including the `SequelizeMeta` and run `npm run migrate` again. because the meta is still reading  old migrations

**Review by timolinn — DISMISSED** [review]

**Review by RaymondMwaura — DISMISSED** [review]
I've run all the tests and it works. Keep up the good work.

**Review by gildniy — DISMISSED** [review]
Tested and works as expected.

**Review by gildniy — APPROVED** [review]
It works, well done!

**Review by RaymondMwaura — APPROVED** [review]
I think this is well implemented. Great job.

**Review by nshaphan — APPROVED** [review]
LGTM

**Review by timolinn — APPROVED** [review]

**Pull request merged**